### ISSUE TITLE
Roll The Dice Getting Started for C++

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/opentelemetry.io#3345]
To complete #2623 we need an updated version of the C++ Getting Started with a "Roll The Dice" app. To simplify things one could use [oatpp](https://oatpp.io/docs/start/) or [drogon](https://github.com/drogonframework/drogon) or any other C++-based framework.

**Comment by Akhaled19** [comment]
Hi @svrnm, I would like to work on this! I just need to figure out a few things

**Comment by svrnm** [comment]
@Akhaled19 that's great to hear, thank you!

To provide you with some help, here are a few pointers:

* Take a look at the [Rust](https://opentelemetry.io/docs/instrumentation/rust/getting-started/) & [Swift](https://opentelemetry.io/docs/instrumentation/swift/getting-started/) getting started, they represent good examples of what we are aiming for as a starting point: provide code for a roll the dice webserver, instrument it, provide pointers to next steps. 
* My own C++ skills are rusty, but I have an oatpp app as a starting point for you, if you like to take that as an example (not a requirement, just a suggestion, use whatever framework suits you):
```cpp
#include "oatpp/web/server/HttpConnectionHandler.hpp"
#include "oatpp/network/Server.hpp"
#include "oatpp/network/tcp/server/ConnectionProvider.hpp"

class Handler : public oatpp::web::server::HttpRequestHandler {
public:
  std::shared_ptr<OutgoingResponse> handle(const std::shared_ptr<IncomingRequest>& request) override {
    std::string response = std::to_string(1 + (std::rand() % 6));
    return ResponseFactory::createResponse(Status::CODE_200, response);
  }
};

void run() {
  auto router = oatpp::web::server::HttpRouter::createShared();
  router->route("GET", "/rolldice", std::make_shared<Handler>());
  auto connectionHandler = oatpp::web::server::HttpConnectionHandler::createShared(router);
  auto connectionProvider = oatpp::network::tcp::server::ConnectionProvider::createShared({"localhost", 8000, oatpp::network::Address::IP_4});
  oatpp::network::Server server(connectionProvider, connectionHandler);
  OATPP_LOGI("MyApp", "Server running on port %s", connectionProvider->getProperty("port").getData());
  server.run();
}

int main() {
  oatpp::base::Environment::init();
  run();
  oatpp::base::Environment::destroy();
  return 0;
}
```

**Comment by Akhaled19** [comment]
@svrnm thank you for the helpful pointers!

**Comment by svrnm** [comment]
@Akhaled19 sure, let us know if you need any more help and don't hesitate to raise your PR early to get feedback!

**Comment by Akhaled19** [comment]
Hi @svrnm I managed to get the app up and running but I'm having a hard time linking OpenTelemetry with my oatpp Application. I'm new to cmake. Here is what my CMakeLists.txt file looks like 
![Screen Shot 2023-10-15 at 5 52 04 PM](https://github.com/open-telemetry/opentelemetry.io/assets/56312360/3077addb-2471-48f6-ba8d-f875ec40bad3)

**Comment by svrnm** [comment]
@Akhaled19, if you share the work you have done so far in textual form (snippets here in the comments or even a github repo!), making it easier for us to take a look into what you have tried to accomplish so far.

@open-telemetry/cpp-approvers

**Comment by svrnm** [comment]
Note that C++11 support has been dropped: https://github.com/open-telemetry/opentelemetry-cpp/pull/2342

**Comment by Akhaled19** [comment]
Got it. I will work on the pr

**Comment by marcalff** [comment]
@svrnm Please close, was fixed by PR #3423
